lib/typed.ml

```diff
@@ -42,21 +42,25 @@ end = struct
 end
 
 and PackageType : sig
-  type t = {type_params: string list;
+  type t = {name : string;
```

Comment:
None of these types should have name fields. I think we can assume that instantiations T(...) only ever have a typename or specialized type for T so we won't need the name.

Reply:
What about serializable enums?

Reply:
I removed the type names that i could